third_party/katana/lib/usdKatana/cache.cpp

```diff
@@ -264,23 +264,21 @@ UsdStageRefPtr UsdKatanaCache::GetStage(
     const std::string contextPath = givenAbsPath ? 
                                     TfGetPathName(fileName) : ArchGetCwd();
 
-    std::string path = !givenAbsPath ? _ResolvePath(fileName) : fileName;
-
     TF_DEBUG(USDKATANA_CACHE_STAGE).Msg(
             "{USD STAGE CACHE} Creating and caching UsdStage for "
             "given filePath @%s@, which resolves to @%s@\n", 
-            fileName.c_str(), path.c_str());
+            fileName.c_str(), _ResolvePath(fileName).c_str());
 
-    if (SdfLayerRefPtr rootLayer = SdfLayer::FindOrOpen(path)) {
+    if (SdfLayerRefPtr rootLayer = SdfLayer::FindOrOpen(fileName)) {
         SdfLayerRefPtr& sessionLayer =
                 _FindOrCreateSessionLayer(sessionAttr, sessionRootLocation);
 
         UsdStageCache& stageCache = UsdUtilsStageCache::Get();
         UsdStageCacheContext cacheCtx(stageCache);
         const UsdStage::InitialLoadSet load = 
             (forcePopulate ? UsdStage::LoadAll : UsdStage::LoadNone);
-        UsdStageRefPtr const& stage = UsdStage::Open(rootLayer, sessionLayer, 
-                ArGetResolver().GetCurrentContext(),
+        UsdStageRefPtr const& stage = UsdStage::Open(rootLayer, sessionLayer,
+                ArGetResolver().CreateDefaultContextForAsset(fileName),
```

Comment:
What was the reason for changing this to use a new context rather than whatever context was currently bound?